grype/matcher/internal/purl.go

```diff
@@ -0,0 +1,77 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/anchore/grype/grype/match"
+	"github.com/anchore/grype/grype/pkg"
+	"github.com/anchore/grype/grype/search"
+	"github.com/anchore/grype/grype/version"
+	"github.com/anchore/grype/grype/vulnerability"
+	"github.com/anchore/packageurl-go"
+)
+
+var ErrEmptyPURLMatch = errors.New("attempted PURL match against package with no PURL")
+
+func MatchPackageByPURL(provider vulnerability.Provider, p pkg.Package, upstreamMatcher match.MatcherType) ([]match.Match, error) {
```

Comment:
I wonder if this would be simpler as a facade over https://github.com/anchore/grype/blob/4b35276c4b064f74df0c637bf60c93fafc6f9934/grype/matcher/internal/language.go#L31 ? 

It looks like its copying some things from the CPE matcher that we don't need, like the confidence of `0.9` and the special JVM version handling.